tests/test.sh

```diff
@@ -0,0 +1,151 @@
+#!/bin/sh
+
+set -x
+DEVFILES_DIR="$(pwd)/stacks/"
+FAILED_TESTS=""
+
+getURLs() {
+    urls=$($ODO_PATH url list | awk '{ print $3 }' | tail -n +3 | tr '\n' ' ')
+    echo "$urls"
+}
+
+# periodicaly check url till it returns expected HTTP status
+# exit after 10 tries
+waitForHTTPStatus() {
+   url=$1
+   statusCode=$2
+
+    for i in $(seq 1 10); do
+        echo "try: $i"
+        content=$(curl -i "$url")
+        echo "Checking if $url is returning HTTP $statusCode"
+        echo "$content" | grep -q -E "HTTP/[0-9.]+ $statusCode"
+        retVal=$?
+        if [ $retVal -ne 0 ]; then
+            echo "ERROR not HTTP $statusCode"
+            echo "$content"
+        else
+            echo "OK HTTP $statusCode"
+            return 0
+        fi
+        sleep 10
+    done
+    return 1
+}
+
+# run test on devfile
+# parameters:
+# - name of a component and project 
+# - path to devfile.yaml
+test() {
+    devfileName=$1
+    devfilePath=$2
+
+    # remember if there was en error
+    error=false
+
+    tmpDir=$(mktemp -d)
+    cd "$tmpDir" || return 1
+
+    $ODO_PATH project create "$devfileName" || error=true
+    if $error; then
+        echo "ERROR project create failed"
+        FAILED_TESTS="$FAILED_TESTS $devfileName"
+        return 1
+    fi
+    
+    $ODO_PATH create "$devfileName" --devfile "$devfilePath" --starter || error=true
+    if $error; then
+        echo "ERROR create failed"
+        $ODO_PATH project delete -f "$devfileName"
+        FAILED_TESTS="$FAILED_TESTS $devfileName"
+        return 1
+    fi
+
+    if [ "$ENV" = "minikube" ]; then
+        # ToDo: Clean up, I'm not happy about having specific checks for the stacks with multiple ports
+        # But since we're testing against minikube, we need to specifically create the URL/ingress before pushing
+        # And if there's multiple ports in the devfile, a port must be specified.
+        if [ "$devfileName" = "java-wildfly" ] || [ "$devfileName" = "java-wildfly-bootable-jar" ]; then
+            $ODO_PATH url create --host "$(minikube ip).nip.io" --port 8080 || error=true
```

Comment:
another alternative is to create a odo url with all the endpoints and checking one of them returns 200 🤔

Reply:
Let me see if that can be quickly adapted, if so I’ll add it.

Reply:
@maysunfaisal Updated.